src/app/examples/todos/todos.reducer.spec.ts

```diff
@@ -0,0 +1,67 @@
+import * as fromTodo from './todos.reducer';
+
+describe('TodoReducer', () => {
+  it('should return the default state', () => {
+    const { initialState } = fromTodo;
+    const action = {} as any;
+    const state = fromTodo.todosReducer(undefined, action);
+
+    expect(state).toBe(initialState);
+  });
+
+  it('should add a todo', () => {
+    const TEST_INITIAL_STATE: fromTodo.TodosState = {
```

Comment:
What are your thoughts about creating a factory function instead of defining `TEST_INITIAL_STATE` in every test?

This would look like this:
```
const createState = ({
  items: [],
  filter: 'ALL',
}) = {}) => ({
  items, 
  filter
});
```

And in the tests:

Reply:
The thing is that if you check all of the other tests, then they have all different state so the default doesn't help that much. 

The factory approach could make sense in case we have a much more complex state but then it would still be probably simpler to do `const state = { ...DEFAULT_STATE, { prop: 'value', otherProp: 'otherValue' } }` with default state defined before tests with `const DEFAULT_STATE = { /* ... */ };`.

I am a big fan of locality and simplicity. Introducing abstraction has sense but it is a tradeoff as everything in engineering. In simple case we're better of without abstractions.